aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,458 @@
+     * @throws JSONException has to be declared as part of creating a test JSON object
+     */
+    @Test(expected = AuthException.class)
+    public void testConfigureExceptionHandling() throws AmplifyException, JSONException {
+        JSONObject pluginConfig = new JSONObject().put("TestKey", "TestVal");
+        JSONObject json = new JSONObject().put("plugins",
+                new JSONObject().put(
+                        new AWSCognitoAuthPlugin().getPluginKey(),
+                        pluginConfig
+                )
+        );
+        AuthCategoryConfiguration authConfig = new AuthCategoryConfiguration();
+        authConfig.populateFromJSON(json);
+
+        doAnswer(invocation -> {
+            Callback<UserStateDetails> callback = invocation.getArgument(2);
+            callback.onError(new Exception());
+            return null;
+        }).when(mobileClient).initialize(any(), any(), any());
+
+        authCategory.configure(authConfig, getApplicationContext());
+    }
+
+    /**
+     * Tests that the signUp method of the Auth wrapper of AWSMobileClient (AMC) calls AMC.signUp() with the username
+     * and password it received and, if included, the userAttributes and validationData sent in the options object.
+     * Also ensures that in the onResult case, the success callback receives a valid AuthSignUpResult.
+     * @throws AuthException test fails if this gets thrown since method should succeed
+     */
+    @Test
+    public void signUp() throws AuthException {
+        SignUpResult amcResult = new SignUpResult(
+            false,
+            new UserCodeDeliveryDetails(
+                    DESTINATION,
+                    DELIVERY_MEDIUM,
+                    ATTRIBUTE_NAME
+            ),
+            null
+        );
+
+        doAnswer(invocation -> {
+            assertEquals(USERNAME, invocation.getArgument(0));
+            assertEquals(PASSWORD, invocation.getArgument(1));
+            Map<String, String> attributeMap = invocation.getArgument(2);
+            assertEquals(ATTRIBUTE_VAL, attributeMap.get(ATTRIBUTE_KEY));
+            assertEquals(METADATA, invocation.getArgument(3));
+
+            Callback<SignUpResult> callback = invocation.getArgument(4);
+            callback.onResult(amcResult);
+            return null;
+        }).when(mobileClient).signUp(any(), any(), any(), any(), any());
+
+        AuthSignUpResult result = synchronousAuth.signUp(
+                USERNAME,
+                PASSWORD,
+                AWSCognitoAuthSignUpOptions.builder()
+                    .userAttributes(
+                        Collections.singletonList(new AuthUserAttribute(AuthUserAttributeKey.email(), ATTRIBUTE_VAL))
+                    )
+                    .validationData(METADATA)
+                    .build()
+        );
+
+        validateSignUpResult(result, AuthSignUpStep.CONFIRM_SIGN_UP_STEP);
+        verify(mobileClient).signUp(any(), any(), any(), any(), any());
```

Comment:
You can probably skip the `doAnswer` stuff, and then just use `eq` to check what is passed in.

Most of these tests should just be like:
```
// Arrange some options foo, bar, baz
Foo foo = new Foo();
Bar bar = mock(Bar.class);
Baz baz = getABaz();

// Act by calling Auth API
auth.signUp(foo, bar, baz, whatever);

// veriify that the mobile client was called in a certain way...
verfiy(mobileClient).signUp(eq(foo), eq(baz));
```

Or, in some cases you might not be able to use`eq`, directly. In which case you can do:
```
auth.signUo("foo", "bar", "baz");

ArgumentCaptor<TypeYouCantMake> whatDidItGiveMe = ArgumentCaptor.forClass(TypeYouCantMake.class);
verify(mobileClient).signUp(whatDidItGiveMe.capture(), eq(bar), eq(baz));

Foo foo = whatDidItGiveMe.value().getSomeContainedField();
assertEquals(expectedFoo, foo); // Ah!!  the thing it gave us _contained_ an expected foo.
```

Reply:
Ok